src/learning/k_means.rs

```diff
@@ -209,14 +209,13 @@ impl<InitAlg: Initializer> KMeansClassifier<InitAlg> {
     /// Used internally within model.
     fn update_centroids(&mut self, inputs: &Matrix<f64>, classes: Vector<usize>) {
         let mut new_centroids = Vec::with_capacity(self.k * inputs.cols());
-        for i in 0..self.k {
-            let vec_i: Vec<usize> = classes.data()
-                .iter()
-                .enumerate()
-                .filter(|&(_, &x)| x == i)
-                .map(|(idx, _)| idx)
-                .collect();
-            
+
+        let mut row_indexes = vec![Vec::new(); self.k];
```

Comment:
Could you explain the motivation for this change? I'm not sure which is better but if you have some insight I'd love to find out!